pkg/ucp/initializer/service.go

```diff
@@ -0,0 +1,135 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package initializer
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/url"
+	"os"
+	"time"
+
+	aztoken "github.com/radius-project/radius/pkg/azure/tokencredentials"
+	"github.com/radius-project/radius/pkg/cli/manifest"
+	"github.com/radius-project/radius/pkg/sdk"
+	"github.com/radius-project/radius/pkg/ucp"
+	"github.com/radius-project/radius/pkg/ucp/api/v20231001preview"
+	"github.com/radius-project/radius/pkg/ucp/hosting"
+
+	//ucpoptions "github.com/radius-project/radius/pkg/ucp/hostoptions"
+	"github.com/radius-project/radius/pkg/ucp/ucplog"
+)
+
+// Service implements the hosting.Service interface for registering manifests.
+type Service struct {
+	options *ucp.Options
+}
+
+var _ hosting.Service = (*Service)(nil)
+
+// NewService creates a server to register manifests.
+func NewService(options *ucp.Options) *Service {
+	return &Service{
+		options: options,
+	}
+}
+
+// Name gets this service name.
+func (s *Service) Name() string {
+	return "initializer"
+}
+
+func waitForServer(ctx context.Context, host, port string, retryInterval time.Duration, connectionTimeout time.Duration, timeout time.Duration) error {
+	address := net.JoinHostPort(host, port)
+	deadline := time.Now().Add(timeout)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("connection attempts canceled or timed out: %w", ctx.Err())
+		default:
+			conn, err := net.DialTimeout("tcp", address, connectionTimeout)
+			if err == nil {
+				conn.Close()
+				return nil
+			}
+
+			if time.Now().After(deadline) {
+				return fmt.Errorf("failed to connect to %s after %v: %w", address, timeout, err)
+			}
+
+			time.Sleep(retryInterval)
+		}
+	}
+}
+
+func (w *Service) Run(ctx context.Context) error {
+	logger := ucplog.FromContextOrDiscard(ctx)
+
+	if w.options.UCP == nil || w.options.UCP.Endpoint() == "" {
+		return fmt.Errorf("connection to UCP is not set")
+	}
+
+	// Parse the endpoint URL and extract host and port
+	parsedURL, err := url.Parse(w.options.UCP.Endpoint())
+	if err != nil {
+		return fmt.Errorf("failed to parse endpoint URL: %w", err)
+	}
+
+	hostName, port, err := net.SplitHostPort(parsedURL.Host)
+	if err != nil {
+		return fmt.Errorf("failed to split host and port: %w", err)
+	}
+	logger.Info("Parsed Host and Port", "host", hostName, "port", port)
+
+	// Attempt to connect to the server
+	err = waitForServer(ctx, hostName, port, 500*time.Millisecond, 500*time.Millisecond, 5*time.Second)
+	if err != nil {
+		logger.Error(err, "Server is not available for manifest registration")
+		return nil
+	}
+
+	// Server is up, proceed to register manifests
+	manifestDir := w.options.Config.Initialization.ManifestDirectory
+	if manifestDir == "" {
+		logger.Info("No manifest directory specified")
```

Comment:
nit: It would be nice to say something like: `No manifest directory specified, initialization is complete.`. That makes it clear that the initializer's job is done.